src/main.rs

```diff
@@ -196,12 +201,60 @@ fn parse<T: BufRead>(
                         );
                     }
-                    TestEvent::Failed { name, stdout } => {
+                    TestEvent::Failed {
+                        name,
+                        stdout,
+                        stderr,
+                    } => {
                         assert!(tests.remove(name));
                         let (name, module_path) = split_name(&name);
+                        let stdout = match stdout {
+                            Some(stdout) => {
+                                let s_data = strip_ansi_escapes::strip(stdout)?;
+                                String::from_utf8_lossy(&s_data).to_string()
+                            }
+                            None => String::new(),
+                        };
+                        let stderr = match stderr {
+                            Some(stderr) => {
+                                let s_data = strip_ansi_escapes::strip(stderr)?;
+                                String::from_utf8_lossy(&s_data).to_string()
+                            }
+                            None => String::new(),
+                        };
+
+                        let system_out = if stderr.is_empty() {
+                            // no stderr ==> return just the stdout
+                            stdout
+                        } else if stdout.is_empty() {
+                            // stderr but no stdout => return just the stderr
+                            stderr
+                        } else {
+                            // both stdout and stderr => return both joined with a newline
+                            format!("{}\n{}", stdout, stderr)
+                        };
+
+                        let system_out = if system_out.len() > max_stdout_len {
+                            format!(
+                                "{}\n[...TRUNCATED...]\n{}",
+                                system_out.split_at(max_stdout_len / 2).0,
+                                system_out
+                                    .split_at(system_out.len() - (max_stdout_len / 2))
+                                    .1
+                            )
+                        } else {
+                            system_out
+                        };
+
                         *current_suite = current_suite.clone().add_testcase(
-                            TestCase::failure(&name, duration, "cargo test", &stdout)
-                                .set_classname(module_path.as_str()),
+                            TestCase::failure(
+                                &name,
+                                duration,
+                                "cargo test",
+                                &format!("failed {}::{}", module_path.as_str(), &name),
+                            )
+                            .set_classname(module_path.as_str())
+                            .set_system_out(&system_out),
```

Comment:
@sasa-tomic Is there a reason that both `stdout` and `stderr` are piped into `set_system_out`?  It seems that routing each to `set_system_out` and `set_system_err` respectively is a better match.

Reply:
Yes, you're right but junit-report does not provide system_err. And we didn't have the time to extend the junit-report.

Reply:
Or hmm looks like there is system_err in junit-report. I was wrong. Either I didn't look well earlier or this was added in the meantime.

Reply:
Would you mind taking over this pr and improving it? I'm quite swamped with other things atm and won't have time to work on this in the next week or two.

Reply:
No problem. Can you make sure that I have permission to push changes to this pr? ￼ https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

Reply:
I believe this is set already, @johnterickson  

![image](https://user-images.githubusercontent.com/34031969/111124664-0b298080-8571-11eb-9891-5f5c88c43015.png)

Reply:
Friendly ping. :) I'd love to see this merged!